src/configuration.ts

```diff
@@ -1507,7 +1500,7 @@ export async function setNewConfiguration(): Promise<void> {
 
         if (configureAfterCommand) {
             logger.message("Automatically reconfiguring the project after a makefile configuration change.");
-            await make.configure(make.TriggeredBy.configureAfterConfigurationChange);
+            await make.cleanConfigure(make.TriggeredBy.configureAfterConfigurationChange);
```

Comment:
Setting a new configuration or a new target requires a clean config versus a non clean config, basically --always-make is needed to recalculate what compilation would look like for all the files even for the up to date ones. Otherwise, the user will see incorrect IntelliSense that slowly updates itself in time as they edit more source files.

Reply:
Seems reasonable enough. We should keep an eye though on the issues list if people notice this causing performance hits.